CodeU_Assignement_1/src/codeu_assignement_1/main.java

```diff
@@ -0,0 +1,278 @@
+        if (args.length!=4)
+        {
+            System.out.println("Error: " + args.length 
+                    + " parameters only. Expect somthing like: Listen Silen 5 1,2,3,4,5,6,7");      
+            return;
+        }
+         
+        // Q1: Run and result
+        System.out.println("Q1: " + args[0] 
+                + (q1_permutation(args[0],args[1]) ? " is " : " isn't ") 
+                + "a permutation of " + args[1]);
+        
+        // Q2: Run and result
+        int k2last = 0;
+        try {
+            k2last = Integer.parseInt(args[2]);
+        } catch (NumberFormatException e) {
+            System.out.println("Error: " + args[2] 
+                    + " is not a number, shall be the k-th to the last location");      
+            return;
+        }
+        
+        // Create a list:
+        String[] list_user = args[3].split(",");
+        SinglyLinkedList<String> intput_list;
+        intput_list = new SinglyLinkedList<>(Arrays.asList(list_user));
+        
+        // Q2: Run and result
+        String res = q2_getKthElement(intput_list, k2last); // Incase of error will throw exception before print
+        System.out.println("Q2: the " + k2last + "-th to the last element is " + res);
+    }
+    
+    /*
+    * Q1: doesn't need a class, a simple method
+    * Input is taken from the main parameters
+    * Output to the screen
+    */
+    static boolean q1_permutation(String s1, String s2) 
+    {
+        // Sanity checks
+        if (s1 == null || s2 == null)
+            if (s1 == null && s2 == null)
+                return true;
+            else 
+                return false;
+        
+        if (s1.length() != s2.length()) return false;
+        if (s1.equals(s2)) return true;
+                
+        /* Start the algorithm : */
+        
+        // To array of chars
+        char[] t1 = s1.toLowerCase().toCharArray();
+        char[] t2 = s2.toLowerCase().toCharArray();
+        
+        // Order
+        Arrays.sort(t1);
+        Arrays.sort(t2);
+        
+        // Compare
+        return Arrays.equals(t2, t1);
+    } // End Q1
+    
+    /*
+     * Works on my own SinglyLinkedList
+     * Index starts from 0 to size-1
+     */
+    static <E> E q2_getKthElement(SinglyLinkedList<E> l, int k2last) 
+    {
+        if (l.isEmpty())
+            throw new IndexOutOfBoundsException(); // Any index is out of bound
+        
+        // Gets the k-th location out-of the k-th to the last location
+        int k = l.size() - k2last - 1;
+        if (k < 0 || k > l.size()) 
+            throw new IndexOutOfBoundsException();
+    
+        return l.get(k);
+    } // End Q2
+    
+    // Some testings to the algorithms
+    static void run_tests_Q1()
+    {
+       String s1 = null;
+       String s2 = null;
+       System.out.println("Q1: (true) " + s1 + (q1_permutation(s1,s2) ? " is " : " isn't ") 
+                + "a permutation of " + s2);
+       
+       s1 = "";

```

Comment:
It's nice to make each test self-enclosed. As it is, right now, this test relies on s2 being set in line 116, as part of the previous test, and s1 being set in line 120.

If you look at an introduction to unit testing, you can see how people try to define small, self-enclosed tests that can be run independently.